modules/liveIntentAnalyticsAdapter.js

```diff
@@ -0,0 +1,157 @@
+  }
+});
+
+liAnalytics.handleAuctionEnd = function(args) {
+  setTimeout(() => {
+    const auction = auctionManager.index.getAuction(args.auctionId);
+    const winningBids = (auction) ? auction.getWinningBids() : [];
+    const data = liAnalytics.createAnalyticsEvent(args, winningBids);
+    sendAnalyticsEvent(data);
+  }, bidWonTimeout);
+}
+
+function getAnalyticsEventBids(bidsReceived) {
+  return bidsReceived.map(bid => {
+    return {
+      adUnitCode: bid.adUnitCode,
+      timeToRespond: bid.timeToRespond,
+      cpm: bid.cpm,
+      currency: bid.currency,
+      ttl: bid.ttl,
+      bidder: bid.bidder
+    };
+  });
+}
+
+liAnalytics.getBannerSizes = function(banner) {
+  if (banner && banner.sizes) {
+    return banner.sizes.map(size => {
+      const [width, height] = size;
+      return {w: width, h: height};
+    });
+  } else return [];
+}
+
+function getUniqueBy(arr, key) {
+  return [...new Map(arr.map(item => [item[key], item])).values()]
+}
+
+liAnalytics.createAnalyticsEvent = function(args, winningBids) {
+  let payload = {}
+  let allUserIds = [];
+
+  payload['instanceId'] = generateUUID();
+  payload['url'] = window.location.href;
+  payload['bidsReceived'] = getAnalyticsEventBids(args.bidsReceived);
+
+  payload['auctionStart'] = args.timestamp;
+  payload['auctionEnd'] = args.auctionEnd;
+
+  payload['adUnits'] = [];
+  payload['userIds'] = [];
+  payload['bidders'] = [];
+
+  if (args.adUnits) {
+    args.adUnits.forEach(unit => {
+      if (unit.mediaTypes && unit.mediaTypes.banner) {
+        payload['adUnits'].push({
+          code: unit.code,
+          mediaType: 'banner',
+          sizes: liAnalytics.getBannerSizes(unit.mediaTypes.banner),
+          ortb2Imp: unit.ortb2Imp
+        });
+      }
+      if (unit.bids) {
+        let userIds = unit.bids.flatMap(getAnalyticsEventUserIds);
+        allUserIds.push(...userIds);
+        let bidders = unit.bids.map(getBidder);
+        payload['bidders'].push(...bidders);
+      }
+    })
+    let uniqueUserIds = getUniqueBy(allUserIds, 'source');
+    payload['userIds'] = uniqueUserIds;
+  }
+  payload['winningBids'] = getAnalyticsEventBids(winningBids);
+  payload['auctionId'] = args.auctionId;
+  return payload;
+}
+
+function getBidder(bid) {
+  return {
+    bidder: bid.bidder,
+    params: bid.params
+  };
+}
+
+function getAnalyticsEventUserIds(bid) {
+  if (bid && bid.userIdAsEids) {
+    return bid.userIdAsEids.map(userId => {
+      let analyticsEventUserId = {
+        source: userId.source,
+        uids: userId.uids,
+        ext: userId.ext
+      };
+      return ignoreUndefined(analyticsEventUserId)
+    });
+  } else { return []; }
+}
+
+function sendAnalyticsEvent(data) {
+  ajax(URL, {
+    success: function () {
+      logInfo('LiveIntent Prebid Analytics: send data success');
+    },
+    error: function (e) {
+      logInfo('LiveIntent Prebid Analytics: send data error' + e);
```

Comment:
Do we want `warn` here? What do the other adapters do?

Reply:
I can update it, there is byDataAnalyticsAdapter it uses logInfo